bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gmos/GmosRecipe.java

```diff
@@ -72,6 +66,18 @@ public GmosRecipe(SourceDefinitionParameters sdParameters,
         _gmosParameters = gmosParameters;
         _teleParameters = teleParameters;
         _plotParameters = plotParameters;
+
+
+        // TODO: this validation should be done centrally somewhere (several instruments do similar stuff here)
+        if (_sdParameters.getDistributionType().equals(SourceDefinitionParameters.Distribution.ELINE)) {
+            if (_sdParameters.getELineWidth() < (3E5 / (_sdParameters.getELineWavelength() * 1000))) {
+                throw new RuntimeException(
+                        "Please use a model line width > 1 nm (or "
+                                + (3E5 / (_sdParameters.getELineWavelength() * 1000))
+                                + " km/s) to avoid undersampling of the line profile when convolved with the transmission response");
+            }
```

Comment:
I appreciate the predicament you’re in with this codebase.  Using exceptions for this kind of thing was probably not the optimal design decision.  These look like input validation and carefully formatted messages for the user of the application. Turning them into unchecked exceptions must mean somewhere at a high level everything is wrapped with a try catch clause and whatever the exception message might be is displayed to the end user?  

If this were new Scala code we would use scalaz’s disjunction or probably `Validation` for this kind of thing.  If it were new code even in Java I’d say we should probably wrap all the return types in something like that.  As is, I’m not sure what to suggest.  So I guess this is just a note to lament the situation we are in with the itc.